### PR TITLE
Gemma4 gguf support

### PR DESCRIPTION
**Pull request by big-yellow-duck**
# Add Gemma 4 GGUF support

## Summary

This PR adds GGUF weight loading support for Gemma 4 text and multimodal models.

- Register a Gemma 4 weights adapter.
- Map text, vision, multimodal projection, router, and expert tensor names from GGUF to the vLLM model layout.
- Split packed MoE gate/up and per-expert tensors into the weights expected by vLLM.
- Reshape the vision patch-embedding weight during loading.
- Add adapter-level coverage and a slow multimodal GGUF integration test using `google/gemma-4-26B-A4B-it` and `unsloth/gemma-4-26B-A4B-it-GGUF:Q4_K_M`.

## Testing

- `pytest -q tests/test_gemma4_adapter.py`
  - 2 passed
- `pytest -vv -s tests/test_multimodal_gguf.py::test_gemma4_mm_gguf`
  - 1 passed in 295.05 seconds on an AMD Radeon 8060S

The integration test validates GGUF loading, multimodal preprocessing and vision encoding, model prefill, and first-token generation against the Hugging Face model.

### online benchmark 

```bash
vllm serve "unsloth/gemma-4-26B-A4B-it-GGUF:UD-Q4_K_M" \
  --tokenizer "google/gemma-4-26B-A4B-it" \
  --served-model-name "google/gemma-4-26B-A4B-it"
```
image

this is benched with guidellm across concurrency 1-8 on the same prefix extending workload.

## Known limitation

Multi-token comparison is currently limited by a Transformers/ROCm issue: the Hugging Face Gemma 4 model produces finite logits for the first generated token, but every logit becomes NaN on the first cached decode step.

This was reproduced with Transformers 5.15.0, PyTorch 2.11.0, and ROCm 7.14. The Gemma 4 integration test therefore compares one generated token. Adapter-level tests separately cover the tensor-name mapping and packed-expert transformation.

### CODE REVIEW

**Review by Isotr0py — APPROVED** [review]
LGTM, thanks!

File: vllm_gguf_plugin/weights_adapter/gemma4.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        orig_to_new_substr=GEMMA4_VISION_SUBSTR,
+    )
+
+
+def _map_tensor_name(mapper: WeightsMapper, name: str) -> str | None:
+    mapped = mapper.apply_list([name])[0]
+    return mapped if mapped != name else None
+
+
+class Gemma4GGUFAdapter(BaseGGUFWeightsAdapter):
```
Comment (Isotr0py) [inline vllm_gguf_plugin/weights_adapter/gemma4.py:100]: BTW, can you also update readme's supported models?

Comment (big-yellow-duck) [reply vllm_gguf_plugin/weights_adapter/gemma4.py:100]: yes I'll update them

**Comment by big-yellow-duck** [comment]
the changes are ready. thanks for the review!
